### PR TITLE
Artem fr

### PR DESCRIPTION
**Pull request by ArtemFrostM**
Added:
- TWO full Dynamic Algorithms classes;
- TWO methods in Algorithms class;
- TWO constants in Dynamic Algorithms enum;
- TWO Switch cases (with breaks :))

### CODE REVIEW

File: src/main/java/com/softserve/algorithms/Algorithms.java [hunk 1]
```diff
@@ -54,4 +55,53 @@ else if (i < m)
...
+    public static int findingMaxIntAlgorithm(int x, int y) {
+		if (x > y) {
+			return x;
+		} else {
+			return y;
+		}
+	}
+    
+    public static BigDecimal findingMaxWaysOfPaintingTheFence(int n, int k) {
+    	if (n <= 0 || k <= 0) {
```
Comment (KatyaKaratnyk) [inline src/main/java/com/softserve/algorithms/Algorithms.java]: Give meaningflul names for variables

**Review by KatyaKaratnyk — CHANGES_REQUESTED** [review]

**Review by KatyaKaratnyk — APPROVED** [review]

**Review by KatyaKaratnyk — APPROVED** [review]

**Review by KatyaKaratnyk — APPROVED** [review]

File: src/main/java/com/softserve/cli/impl/MaxPossibleDailyTasksCLI.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		this.low = ArrayUtil.parseIntegerSequence(sequence);
+		
+		if(this.checkIfArrHasNegativeValues(this.low) == false) {
+			return false;
+		}
+		
+		return (this.low.length == this.n);
+	}
+
+	private void comparingArrs(int arrHigh[], int arrLow[]) {
```
Comment (igorflyunt) [inline src/main/java/com/softserve/cli/impl/MaxPossibleDailyTasksCLI.java]: Logic of this method belongs in the `findindMaxSumOfPossibleDailyTasks` method

File: src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	@Override
+	public void run() {
+		System.out.printf("%s\n%s\n", algorithmOption.getOptionName(),
+                algorithmOption.getOptionDescription());
+        InvalidInputHandler.promptIfInvalidValue(this::promptNParamethers, INVALID_NUMBER_MSG);
+        InvalidInputHandler.promptIfInvalidValue(this::promptKParamethers, INVALID_NUMBER_MSG);
+        System.out.println("The MAX number of possible ways of painting the fence is " + Algorithms.findingMaxWaysOfPaintingTheFence(numberOfPosts, numberOfColors));
+	}
+	
+	private boolean promptNParamethers() {
```
Comment (igorflyunt) [inline src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java]: Rename this method to promptNumberOfColors

File: src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        System.out.print("Please enter number of colors (it has to be more than 0): ");
+        if (!InputUtil.hasNextInt(prompt))
+            return false;
+        this.numberOfPosts = InputUtil.readIntValue(prompt);
+        return this.numberOfPosts>=1;
+    }
+
+    private boolean promptKParamethers() {
```
Comment (igorflyunt) [inline src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java]: Rename this method to promptNumberOfPosts

File: src/main/java/com/softserve/algorithms/Algorithms.java [hunk 5]
```diff
@@ -54,4 +55,53 @@ else if (i < m)
             return count[n];
         }
     }
+    
+    public static int findindMaxSumOfPossibleDailyTasks(int high[], int low[], int number) {
+		if (number < 1) {
+			throw new NegativeArraySizeException();
```
Comment (igorflyunt) [inline src/main/java/com/softserve/algorithms/Algorithms.java:61]: Add `throws` clause.

File: src/main/java/com/softserve/algorithms/Algorithms.java [hunk 6]
```diff
@@ -54,4 +55,53 @@ else if (i < m)
...
+    
+    public static int findingMaxIntAlgorithm(int firstNumber, int secondNumber) {
+		if (firstNumber > secondNumber) {
+			return firstNumber;
+		} else {
+			return secondNumber;
+		}
+	}
+    
+    public static BigDecimal findingMaxWaysOfPaintingTheFence(int numberOfPosts, int numberOfColors) {
```
Comment (igorflyunt) [inline src/main/java/com/softserve/algorithms/Algorithms.java]: Add `throws` clause.

**Review by igorflyunt — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/algorithms/Algorithms.java (hunk 5 above)
Comment (ArtemFrostM) [reply src/main/java/com/softserve/algorithms/Algorithms.java:61]: > Add `throws` clause.

solved

File: src/main/java/com/softserve/cli/impl/MaxPossibleDailyTasksCLI.java (hunk 2 above)
Comment (ArtemFrostM) [reply src/main/java/com/softserve/cli/impl/MaxPossibleDailyTasksCLI.java]: > Logic of this method belongs in the `findindMaxSumOfPossibleDailyTasks` method

solved

File: src/main/java/com/softserve/algorithms/Algorithms.java (hunk 1 above)
Comment (ArtemFrostM) [reply src/main/java/com/softserve/algorithms/Algorithms.java]: > Give meaningflul names for variables

solved

File: src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java (hunk 3 above)
Comment (ArtemFrostM) [reply src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java]: > Rename this method to promptNumberOfColors

solved

File: src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java (hunk 4 above)
Comment (ArtemFrostM) [reply src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java]: > Rename this method to promptNumberOfPosts

solved

File: src/main/java/com/softserve/algorithms/Algorithms.java (hunk 5 above)
Comment (ArtemFrostM) [reply src/main/java/com/softserve/algorithms/Algorithms.java:61]: > Add `throws` clause.

solved

File: src/main/java/com/softserve/algorithms/Algorithms.java (hunk 6 above)
Comment (ArtemFrostM) [reply src/main/java/com/softserve/algorithms/Algorithms.java]: > Add `throws` clause.

solved

**Review by igorflyunt — APPROVED** [review]

**Pull request merged**